### PR TITLE
usm: http2: Handle frame header remainder in the first packet

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Handling an edge case, where we have a remainder in the first frame, we consume it, and then we assumed (wrongfully) that we have a full valid frame. But we can have a part of a frame thus we need to create a new frame remainder for header remainder

### Motivation

Improve incorrect parsing and accuracy gaps.

### CODE REVIEW

File: pkg/network/ebpf/c/protocols/http2/decoding.h [hunk 1]
```diff
@@ -539,6 +542,18 @@ int socket__http2_handle_first_frame(struct __sk_buff *skb) {
     }
 
     if (!has_valid_first_frame) {
+        if (dispatcher_args_copy.skb_info.data_off < dispatcher_args_copy.skb_info.data_end && dispatcher_args_copy.skb_info.data_off + HTTP2_FRAME_HEADER_SIZE > dispatcher_args_copy.skb_info.data_end) {
+            frame_header_remainder_t new_frame_state = { 0 };
+            // We have a frame header remainder
```
Comment (amitslavin) [inline pkg/network/ebpf/c/protocols/http2/decoding.h:547]: lets make better documentation to make it more clear

File: pkg/network/usm/usm_http2_monitor_test.go [hunk 2]
```diff
@@ -1184,6 +1184,36 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
 			},
 			expectedEndpoints: nil,
 		},
+		{
+			name: "remainder + header remainder",
+			// The purpose of this test is to validate that we cannot handle reassembled tcp segments.
```
Comment (amitslavin) [inline pkg/network/usm/usm_http2_monitor_test.go:1189]: correct the doc

**Review by amitslavin — CHANGES_REQUESTED** [review]

File: pkg/network/usm/usm_http2_monitor_test.go (hunk 2 above)
Comment (guyarb) [reply pkg/network/usm/usm_http2_monitor_test.go:1189]: done

File: pkg/network/ebpf/c/protocols/http2/decoding-tls.h [hunk 3]
```diff
@@ -634,6 +637,18 @@ int uprobe__http2_tls_handle_first_frame(struct pt_regs *ctx) {
         bpf_map_delete_elem(&http2_remainder, &dispatcher_args_copy.tup);
     }
     if (!has_valid_first_frame) {
+        if (dispatcher_args_copy.data_off < dispatcher_args_copy.data_end && dispatcher_args_copy.data_off + HTTP2_FRAME_HEADER_SIZE > dispatcher_args_copy.data_end) {
+            frame_header_remainder_t new_frame_state = { 0 };
+            // We have a frame header remainder
```
Comment (amitslavin) [inline pkg/network/ebpf/c/protocols/http2/decoding-tls.h:642]: change it as you did it in the plain

**Review by amitslavin — APPROVED** [review]

**Pull request merged**